frontend/src/hooks/useUrlQueryData.ts

```diff
@@ -21,11 +22,19 @@ const useUrlQueryData = <T>(
 		(newQueryData: T): void => {
 			const newQuery = JSON.stringify(newQueryData);
 
-			urlQuery.set(queryKey, newQuery);
-			const generatedUrl = `${location.pathname}?${urlQuery.toString()}`;
+			// Create a new URLSearchParams object with the current URL's search params
+			// This ensures we're working with the most up-to-date URL state
+			const currentUrlQuery = new URLSearchParams(window.location.search);
+
+			// Update or add the specified query parameter with the new serialized data
+			currentUrlQuery.set(queryKey, newQuery);
+
+			// Construct the new URL by combining the current pathname with the updated query string
+			const generatedUrl = `${location.pathname}?${currentUrlQuery.toString()}`;
+
 			history.replace(generatedUrl);
 		},
-		[history, location, urlQuery, queryKey],
+		[history, location.pathname, queryKey],
```

Comment:
curious that even the presence if urlQuery in the dependency of `useCallback` isn't letting it to get updated value ?

Reply:
Yeah, it wasn't working even if urlQuery is in the dependencies of useCallback.

The reason is that the `urlQuery` object is created using `useMemo` in `useUrlQuery`, which means it retains the same reference unless the search value changes. Including `urlQuery` in the dependency array of `useCallback` does not trigger a re-evaluation of the callback unless the reference itself changes. By directly using new `URLSearchParams(window.location.search)` in the `redirectWithQuery` function, we ensure that we always work with the latest URL state, avoiding any stale closures.